topdown/query.go

```diff
@@ -291,6 +292,11 @@ func (q *Query) WithStrictObjects(yes bool) *Query {
 	return q
 }
 
+func (q *Query) WithVirtualCache(vc VirtualCache) *Query {
```

Comment:
I think a doc comment here for this new public function would be helpful, it's less self explanatory than `PrintHook` which doesn't have one...